test/Shared/UserManagerTestBase.cs

```diff
@@ -1093,39 +1097,44 @@ public Task<IdentityResult> ValidateAsync(UserManager<TUser> manager, TUser user
         public async Task CanQueryableRoles()
         {
             var manager = CreateRoleManager();
-            var roles = GenerateRoles("CanQuerableRolesTest", 4);
-            foreach (var r in roles)
+            if (manager.SupportsQueryableRoles)
             {
-                IdentityResultAssert.IsSuccess(await manager.CreateAsync(r));
+                var roles = GenerateRoles("CanQuerableRolesTest", 4);
+                foreach (var r in roles)
+                {
+                    IdentityResultAssert.IsSuccess(await manager.CreateAsync(r));
+                }
+                Assert.Equal(roles.Count, manager.Roles.Count(RoleNameStartsWithPredicate("CanQuerableRolesTest")));
+                var r1 = manager.Roles.FirstOrDefault(RoleNameStartsWithPredicate("CanQuerableRolesTest1"));
+                Assert.Equal(roles[1], r1);
```

Comment:
Can we have a negativeassert as in the Users test and standardize on the same test data generation pattern?